src/probabilities_estimators/histograms/visitation_frequency.jl

```diff
@@ -0,0 +1,54 @@
+export ValueHistogram, VisitationFrequency, AbstractBinning
+
+"""
+    AbstractBinning
+
+The supertype of all binning schemes.
+"""
+abstract type AbstractBinning end
+
+"""
+    AbstractBinEncoder
+
+The supertype of all binning encoders, which are used with `symbolize` to "encode"
+a data point into its bin. The function `bin_encoder(x, b)` takes in an AbstractBinning
+and returns the encoder.
+"""
+abstract type AbstractBinEncoder end
+
+
+"""
+    ValueHistogram(b::AbstractBinning) <: ProbabilitiesEstimator
+
+A probability estimator based on binning the values of the data as dictated by
+the binning scheme `b` and formally computing their histogram, i.e.,
+the frequencies of points in the bins. Alias to this is `VisitationFrequency`.
+
+This method has a linearithmic time complexity (`n log(n)` for `n = length(x)`)
+and a linear space complexity (`l` for `l = dimension(x)`).
+This allows computation of probabilities (histograms) of high-dimensional
+datasets and with small box sizes `ε` without memory overflow and with maximum performance.
+To obtain the bin information along with the probabilities, use [`binhist`](@ref).
+
+See also: [`RectangularBinning`](@ref).
+"""
+struct ValueHistogram{RB<:AbstractBinning} <: ProbabilitiesEstimator
+    binning::RB
+end
+
+"""
+    VisitationFrequency
```

Comment:
Blank line after code signature, for consistency with the remaining code after the Markdown cleanup